v2/sender.go

```diff
@@ -29,6 +30,7 @@ type AzServiceBusSender interface {
 // Sender contains an SBSender used to send the message to the ServiceBus queue and a Marshaller used to marshal any struct into a ServiceBus message
 type Sender struct {
 	sbSender AzServiceBusSender
+	mu       sync.RWMutex
```

Comment:
document what the mutex is for, when to take RLock and when to take Lock.
I think, in essence, all usages of sbSender should use RLock and ensure to always release the lock.
the only place to use Lock is in the Failover func.